doc/netplan.md

```diff
@@ -340,6 +340,90 @@ These options are available for all types of interfaces.
      :    Match this policy rule based on the type of service number applied to
           the traffic.
 
+## Authentication
+Netplan supports advanced authentication settings for ethernet and wifi
+interfaces, as well as individual wifi networks, by means of the ``auth`` block.
+
+``auth`` (mapping)
+
+:    Specifies authentication settings for a device of type ``ethernets:`` or
+     ``wifis:``, or a single wifi ``access-points:`` entry. If ``auth`` blocks
+     are specified on both a wifi device and one of its access points, only the
+     settings in the access point's ``auth`` block are taken into account;
+     however, the ``auth`` block of the wifi device, if present, is used for any
+     ``access-points:`` without their own ``auth`` block.
+
+     Example:
+
+          wifis:
+            wlp1s0:
+              auth:
+                key-mgmt: wpa-psk
+                psk: "d3f4ul7"
+              access-points:
+                office:
+                  auth:
+                    # the authentication settings above are ignored for the
+                    # wifi named "office"
+                    key-mgmt: wpa-eap
+                    eap-method: ttls
+                    anonymous-identity: "@internal.example.com"
+                    identity: "joe@internal.example.com"
+                    password: "v4ryS3kr1t"
+                home:
+                  mode: infrastructure
+                  # the authentication settings above are applied to the wifi
+                  # named "home"
+
+      The ``auth`` block supports the following properties:
+
+      ``key-mgmt`` (scalar)
+      :    The supported key management modes are ``none`` (no key management),
+           ``wpa-psk`` (WPA with pre-shared key, common for home wifi),
+           ``wpa-eap`` (WPA with EAP, common for enterprise wifi) and ``8021x``
+           (used primarily for wired Ethernet connections).
+
+      The following properties can be used if ``key-mgmt`` is ``wpa-psk``:
+
+      ``psk`` (scalar)
+      :    The pre-shared key for this connection. Useful only if ``key-mgmt``
+           is ``wpa-psk``.
```

Comment:
what is the rationale for using a different key name for wpa-psk vs other auth types? "password" and "pre-shared key" are synonymous, and used interchangeably in the wild; I don't think it matters that wpa-psk doesn't have an a username associated with the password, and that we should still just call it 'password'.

Reply:
+1, for consistency and because "password" is more human-parseable than "psk", but I do understand the logic of being able to map 1:1 between configuration parameters for wpasupplicant/NM.

WPA personal has "password" being actually mapped to "psk" in supplicant config. "password" here is convenient to remember / easier for a human to map to what it's used for.

WPA enterprise / EAP has "password" being an actual password for the username for the connection.

Reply:
+1, let's use "password" everywhere unless it conflicts with something else that is more obviously a password.

Reply:
I guess it boils down to whether netplan should provide a straightforward mapping between its own options and wpa_supplicant/NM options (although we can certainly change some of the option names), or if there should be some reinterpretation (i.e. knowing whether `password` should be passed as a PSK or as an EAP password depending on `key-mgmt`).

For what it's worth, the [current sample wpa_supplicant.conf](https://w1.fi/cgit/hostap/tree/wpa_supplicant/wpa_supplicant.conf?id=0fe3ede0a4f6fe984bbaad90f1a9d9e8decc155d#n1708) allows both to be set simultaneously, supplying one or the other depending on which authentication mechanism is announced by the other end. However, wpa_supplicant also allows setting multiple values for key_mgmt, picking whichever one the other side claims to support; I don't think making this functionality available in netplan makes much sense.

Reply:
The guiding vision here is that we should learn from and be informed by prior art in naming, but not be constrained by it when it's not user-friendly.  In the short term, there will always be people familiar with the old thing who are dissatisfied with a name change.  In the long term, these people will be outnumbered by others who are learning it new.

netplan.yaml is intended to model network configuration - not any particular backend implementation.